### DISCUSSION TITLE
psychopg2 complaining that a table doesn't exist after migrations

### DISCUSSION THREAD

**Discussion by andre-qumulo** [sqlalchemy/alembic#1473] [Usage Questions] [answered]
I am migrating a SqlAlchemy 1.3 project with Alembic 1.4.2 to SqlAlchemy 2.x and Alembic 1.13.1

I am running into a very confusing problem where when I run a `db.session.execute(text('INSERT INTO ...` into a table that is dropped and then recreated in a migration I get an error that `sqlalchemy.exc.ProgrammingError: (psycopg2.errors.UndefinedTable) relation "some_table_name" does not exist`.

I am trying to understand what to try to try and figure out the issue.

More details are here: https://stackoverflow.com/questions/78481263/sqlalchemy-relation-does-not-exist-after-flask-migrate

**Comment by CaselIT** [comment]
Hi,

Well first thing to try would be to verify it the table is indeed there using psql of another similar tool

**Reply by andre-qumulo** [reply]
These are postgresql dbs that we're creating using pg_ctl in a python in-memory file system so I can't log in via psql

**Reply by CaselIT** [reply]
I'm not familiar with those, but that sounds like a crucial thing to mention.

If they are local to a single connection, like sqlite, then that's the issue. Alembic uses one connection, while your sqlalchemy uses another

**Reply by andre-qumulo** [reply]
If you look at the link above, you can see that we turn on SQLALCHEMY_ECHO and we are seeing `CREATE TABLE` sql statements being ran on the database.

I'll look into the connections

**Reply by CaselIT** [reply]
sure, that seems like two different connections are used.
Try using a real db and everything should work. I'm not aware of pg supporting in memory dbs

edit: oh you edited your description. No clue what you are doing, but it's very likely that's the issue. Try ensuring that your alembic is using the same connection as the rest of the sw

**Reply by andre-qumulo** [reply]
I'm wondering if it's a problem with my env.py file.

We had a file that looked similar to this: https://gist.github.com/sairamkrish/e583abdf7c21aed987754f9a2e252cbd
but I'm seeing `TypeError: alembic.context.configure() got multiple values for keyword argument 'render_as_batch'` errors when I change the arguments to `context.configure(` between the run_migrations_offline and run_migrations_online functions.

Is there an example of what it should look like now?

**Reply by andre-qumulo** [reply]
I was able to get into the DB via psql after the migration should have ran (it looks like we initialize a local host) and `\d` shows no relations.  Do you think this is a problem with Alembic or flask-migrate that we're using?

**Reply by CaselIT** [reply]
I would start by using a normal postgresql configuration.
If that works then try to understand what are the limitations of your custom setup and do the appropriate changes.

I'm also not sure why you are trying to use alembic with a temporary volatile database, it may be easier to just recreate the schema from scratch using `metadata.create_all`

**Reply by andre-qumulo** [reply, marked as answer]
So these are unit/integration tests that allow us to run the migration against a temporary DB to ensure it won't destroy our production one.
I have been able to use `psql` to log into the DB as it's running on localhost.  


Somehow I fixed this by replacing our env.py file with the one from flask-migrate's repo and removing any references to running the migrations as sql
